src/Map.fs

```diff
@@ -11,50 +11,101 @@ type IMap =
-    
+
+    let MapType =
+        Attributes.defineEnum<MapType> "Map_MapType" (fun _ newValueOpt node ->
+            let map = node.Target :?> Map
+
+            let value =
+                match newValueOpt with
+                | ValueNone -> MapType.Street
+                | ValueSome v -> v
+
+            map.MapType <- value)
+
+    let IsShowingUser = Attributes.defineBindableBool Map.IsShowingUserProperty
+
+    let TrafficEnabled = Attributes.defineBindableBool Map.TrafficEnabledProperty
+
     let HasScrollEnabled = Attributes.defineBindableBool Map.HasScrollEnabledProperty
+
     let HasZoomEnabled = Attributes.defineBindableBool Map.HasZoomEnabledProperty
-    let IsShowingUse = Attributes.defineBindableBool Map.IsShowingUserProperty
-    
-    let TrafficEnabledProperty = Attributes.defineBindableBool Map.TrafficEnabledProperty
-    
-    let MoveToLastRegionOnLayoutChange = Attributes.defineBindableBool Map.MoveToLastRegionOnLayoutChangeProperty
-    
-    let MapType =
-        Attributes.defineEnum<MapType>
-            "Map_MapType"
-            (fun _ newValueOpt node ->
-                let map = node.Target :?> Map
-
-                let value =
-                    match newValueOpt with
-                    | ValueNone -> MapType.Street
-                    | ValueSome v -> v
-
-                map.MapType <- value)
-            
-    let ItemsSource<'T> =
-        Attributes.defineBindable<WidgetItems<'T>, System.Collections.Generic.IEnumerable<Widget>>
-            Map.ItemsSourceProperty
-            (fun modelValue ->
-                seq {
-                    for x in modelValue.OriginalItems do
-                        modelValue.Template x
-                })
-            (fun a b -> ScalarAttributeComparers.equalityCompare a.OriginalItems b.OriginalItems)
 
+    let MoveToLastRegionOnLayoutChange =
+        Attributes.defineBindableBool Map.MoveToLastRegionOnLayoutChangeProperty
+
+    let RequestedRegion =
+        Attributes.defineSimpleScalarWithEquality<MapSpan> "Map_RequestedRegion" (fun _ newValueOpt node ->
+            let map = node.Target :?> Map
+
+            match newValueOpt with
+            | ValueNone -> ()
+            | ValueSome mapSpan -> map.MoveToRegion(mapSpan))
+
+    let Pins =
+        Attributes.defineListWidgetCollection "Map_Pins" (fun target -> (target :?> Map).Pins)
+
+    let MapClicked =
+        Attributes.defineEvent<MapClickedEventArgs> "Map_MapClicked" (fun target -> (target :?> Map).MapClicked)
 
 [<AutoOpen>]
 module MapBuilders =
     type Fabulous.XamarinForms.View with
 
         /// Defines a Map widget
-        static member inline Map<'msg>() =
-            WidgetBuilder<'msg, IMap>(Map.WidgetKey, AttributesBundle(StackList.empty (), ValueNone, ValueNone))
+        static member inline Map<'msg>(requestedRegion: MapSpan) =
+            CollectionBuilder<'msg, IMap, IPin>(Map.WidgetKey, Map.Pins, Map.RequestedRegion.WithValue(requestedRegion))
```

Comment:
I'm wondering if this is the right approach.
The issue with CollectionBuilder is that it requires at least one pin to be able to compile, except not everyone wants pins.

Maybe it would be better keeping the builder as WidgetBuilder and have a separate collection modifier named `pins`?

Reply:
Humm Yeah that is a good point .  Will update it :)

Reply:
Updated